packages/flutter/test/widgets/scrollable_test.dart

```diff
@@ -1427,6 +1427,36 @@ void main() {
     expect(syntheticScrollableNode!.hasFlag(ui.SemanticsFlag.hasImplicitScrolling), isTrue);
     handle.dispose();
   });
+
+  testWidgets('Scroll inertia cancel event', (WidgetTester tester) async {
+    // First try without cancelling.
+    await pumpTest(tester, null);
+    await tester.fling(find.byType(Scrollable), const Offset(0.0, -dragOffset), 1000.0);
+    expect(getScrollOffset(tester), dragOffset);
+    await tester.pump(); // trigger fling
+    expect(getScrollOffset(tester), dragOffset);
+    await tester.pump(const Duration(seconds: 5));
+    final double withoutCancelResult = getScrollOffset(tester);
+    resetScrollOffset(tester);
+
+    // Now cancel partway into the inertia.
+    await pumpTest(tester, null);
+    await tester.fling(find.byType(Scrollable), const Offset(0.0, -dragOffset), 1000.0);
+    expect(getScrollOffset(tester), dragOffset);
+    await tester.pump(); // trigger fling
+    expect(getScrollOffset(tester), dragOffset);
+    await tester.pump(const Duration(milliseconds: 200));
+    final TestPointer testPointer = TestPointer(1, ui.PointerDeviceKind.mouse);
+    testPointer.hover(tester.getCenter(find.byType(Scrollable)));
+    await tester.sendEventToBinding(testPointer.scrollInertiaCancel());
+    await tester.pump();
+    final double atCancelResult = getScrollOffset(tester);
+    await tester.pump(const Duration(milliseconds: 4800)); // Add up to the same 5 seconds as before.
+    final double withCancelResult = getScrollOffset(tester);
+
+    expect(withCancelResult, lessThan(withoutCancelResult)); // Inertia was stopped before reaching full distance
```

Comment:
Can we just record the exact result (and possibly remove the first try)?

Reply:
It's dependent on exact scroll friction calculation. I hadn't recorded the exact result since I am working on some related change (the scroll friction right now has been tuned for touch, it causes much more momentum than native trackpad scrolling apps). But I can change it to the exact result now and change again later.